fenix/app/src/main/java/org/mozilla/fenix/home/topsites/TopSites.kt

```diff
@@ -240,25 +249,72 @@ private fun TopSiteFaviconCard(topSite: TopSite) {
                 shape = RoundedCornerShape(4.dp),
             ) {
                 val drawableForUrl = getDrawableForUrl(topSite.url)
+                when {
+                    drawableForUrl != null -> {
+                        FaviconDrawable(drawableForUrl)
+                    }
 
-                if (drawableForUrl != null) {
-                    Image(
-                        painter = painterResource(id = drawableForUrl),
-                        contentDescription = null,
-                        modifier = Modifier.size(TOP_SITES_FAVICON_SIZE.dp),
-                        contentScale = ContentScale.Crop,
-                    )
-                } else {
-                    Favicon(
-                        url = topSite.url,
-                        size = TOP_SITES_FAVICON_SIZE.dp,
-                    )
+                    topSite is TopSite.Provided -> {
+                        FaviconBitmap(topSite)
+                    }
+
+                    else -> {
+                        FaviconDefault(topSite.url)
+                    }
                 }
             }
         }
     }
 }
 
+@Composable
+private fun FaviconDrawable(@DrawableRes drawableForUrl: Int) {
```

Comment:
💭  thoughts on directly calling the `FaviconImage` and removing this function altogether?